src/codegen.jl

```diff
@@ -50,12 +50,15 @@ function augment_impl(var_offset::Int, op_offset::Int, head::DataType, tail::NTu
         num_affine, after_affine = uses_affinemap(head, tail) ? seek_connected(uses_affinemap, 0, head, tail) : (0, nothing)
         num_special, _ = seek_connected(x->(supports_permute(x)||supports_view(x)||supports_stepview(x)), 0, head, tail)
         num_lazy, after_lazy = seek_connected(supports_lazy, 0, head, tail)
+        @assert num_special <= num_affine <= num_lazy # issue #40
         if num_special >= num_affine
```

Comment:
A subtlety here:

```suggestion
         if num_special > num_affine
```

Here's the reasoning behind: when `num_special == num_affine`, according to #40 all these operations support affinemap, hence they can be unrolled using `unroll_applyaffine`

Tests and benchmarks are required to verify this change.